### PR TITLE
updates to use async methods on vector store integrations + aadd & qu…

### PR DESCRIPTION
**Pull request by sourabhdesai**
…ery implementation of those functions for pgvector

# Description

Adding async support for `PGVectorStore` along with changes needed on the parent class & the `VectorIndexRetriever`.

## Type of Change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Added new unit/integration tests
- [x] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: llama_index/vector_stores/postgres.py [hunk 1]
```diff
@@ -101,75 +112,108 @@ def _create_extension(self) -> None:
...
-                            remove_text=True,
-                            flat_metadata=self.flat_metadata,
-                        ),
-                    )
+                    item = self._node_to_table_row(result)
                     session.add(item)
                 session.commit()
         return ids
 
+    async def aadd(self, embedding_results: List[NodeWithEmbedding]) -> List[str]:
```
Comment (sourabhdesai) [inline llama_index/vector_stores/postgres.py:138]: Does `aadd` sound as awkward to you as it does to me? I was trying to follow the existing naming convention with the way we're distinguishing sync vs. async functions but I wouldn't mind renaming this to `async_add`

File: llama_index/vector_stores/types.py [hunk 2]
```diff
@@ -158,15 +158,44 @@ def add(
         """Add embedding results to vector store."""
         ...
 
+    async def aadd(
```
Comment (logan-markewich) [inline llama_index/vector_stores/types.py:161]: My brain is screaming looking at this function name LOL but I got nothing better to suggest

File: llama_index/vector_stores/postgres.py [hunk 3]
```diff
@@ -101,75 +112,108 @@ def _create_extension(self) -> None:
...
+    def _db_rows_to_query_result(self, rows: List[DBEmbeddingRow]) -> VectorStoreQueryResult:
         nodes = []
         similarities = []
         ids = []
-        for item, sim in results:
+        for db_embedding_row in rows:
             try:
-                node = metadata_dict_to_node(item.metadata_)
-                node.set_content(str(item.text))
+                node = metadata_dict_to_node(db_embedding_row.metadata)
```
Comment (logan-markewich) [inline llama_index/vector_stores/postgres.py:229]: Should it be `metadata_` like the old version?

**Review by logan-markewich — APPROVED** [review]
I think this looks good? Maybe double check that the example notebook still works 👌

File: llama_index/vector_stores/postgres.py (hunk 1 above)
Comment (logan-markewich) [reply llama_index/vector_stores/postgres.py:149]: Ha I left the same comment as I was going through this.

I think async_add can be an exception to our naming scheme, aadd just looks like a typo imo

File: llama_index/vector_stores/types.py (hunk 2 above)
Comment (sourabhdesai) [reply llama_index/vector_stores/types.py:161]: cool so its not just me 😂 perhaps `async_add`?

**Review by logan-markewich — APPROVED** [review]
So a user has to manually call async_add to get the async functionality when building an index right? 

Future Todo: add async index creation at the top level 👌

**Pull request merged**